### PR TITLE
chore: Add new codecov components

### PR DESCRIPTION
**Pull request by eirikurn**
## What

Define new codecov components for microservices, auth services, react libs, nest libs and island-ui.

## Why

To track coverages over time for our main test-based focus.

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

**Comment by pshomov** [comment]
I think if you maybe make a dummy change that would trigger a unit test, then all the Codecov components would show up.

File: codecov.yml [hunk 1]
```diff
@@ -18,12 +18,24 @@ component_management:
...
         - apps/api/**
         - libs/api/domains/**
-    - component_id: application-system-api-graphql # this is an identifier that should not be changed
+    - component_id: microservices
+      name: All REST APIs except Application System API.
+      paths:
+        - apps/services/**
+        - apps/air-discount-scheme/backend/**
+        - apps/download-service/**
+        - libs/auth-api-lib/**
```
Comment (saevarma) [inline codecov.yml:32]: We talked about when rethinking the folder structure of our repo that if service requires a library it should have `libs/services/<app name>` to we want to list that path as well? 
Today it is only us using it for shared testing utilities for auth services.

**Review by saevarma — APPROVED** [review]
LGTM
Removed `automerge` so you can review my comment before you decide to merge this.

File: codecov.yml (hunk 1 above)
Comment (eirikurn) [reply codecov.yml:32]: Good point

**Pull request merged**